control-base/src/main/java/fi/nls/oskari/control/layer/GetMapLayerGroupsHandler.java

```diff
@@ -0,0 +1,117 @@
+package fi.nls.oskari.control.layer;
+
+import static fi.nls.oskari.control.ActionConstants.PARAM_LANGUAGE;
+import static fi.nls.oskari.control.ActionConstants.PARAM_SRS;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import fi.mml.map.mapwindow.service.db.OskariMapLayerGroupService;
+import fi.mml.map.mapwindow.service.db.OskariMapLayerGroupServiceIbatisImpl;
+import fi.mml.map.mapwindow.util.OskariLayerWorker;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.domain.map.MaplayerGroup;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.ResponseHelper;
+
+/**
+ * Get all map layer groups registered in Oskari database
+ */
+@OskariActionRoute("GetHierarchicalMapLayerGroups")
+public class GetMapLayerGroupsHandler extends ActionHandler {
+	
+	private static Logger log = LogFactory.getLogger(GetMapLayerGroupsHandler.class);
+
+	private static final String KEY_GROUPS = "groups";
+
+    @Override
+    public void handleAction(ActionParameters params) throws ActionException {
+
+        final String lang = params.getHttpParam(PARAM_LANGUAGE, params.getLocale().getLanguage());
+
+        log.debug("Getting layer groups");
+        OskariMapLayerGroupService groupService = new OskariMapLayerGroupServiceIbatisImpl();
```

Comment:
We shouldn't create new instances of any services we need for every request. This should be moved to a private field of this class, and it should be initialized inside the init() function of the ActionHandler, just like you've done in the `LayerAndGroupOrderHandler`.